pylint/config/option_manager_mixin.py

```diff
@@ -81,7 +81,7 @@ def reset_parsers(self, usage=""):
     def register_options_provider(self, provider, own_group=True):
         """register an options provider"""
         assert provider.priority <= 0, "provider's priority can't be >= 0"
-        for i in range(len(self.options_providers)):
+        for i, _ in enumerate(self.options_providers):
             if provider.priority > self.options_providers[i].priority:
                 self.options_providers.insert(i, provider)
                 break
```

Comment:
```suggestion
        for i, options_provider in enumerate(self.options_providers):
            if provider.priority > options_provider.priority:
                self.options_providers.insert(i, provider)
                break
```

Reply:
oops, oversight on my part -- will accept this suggestion. Thanks!

Reply:
We can probably add a new checker for it ;)
Something like `list-index-lookup`?

Reply:
sounds good, was thinking something along the lines too